assemblyscript/assembly/as-wasi-nn.ts

```diff
@@ -17,8 +17,8 @@ import * as wasi_ephemeral_nn from './wasi_ephemeral_nn';
 
 function getArrayPtr<T>(data: T): u32 {
     // Only typed arrays have byteOffset. Cast to typed.
-    let u32Data = Uint32Array.wrap(data.buffer);
-    return (changetype<u32>(u32Data.buffer) + u32Data.byteOffset);
+    let u8Data = Uint8Array.wrap(data.buffer);
+    return (changetype<u32>(u8Data.buffer) + u8Data.byteOffset);
```

Comment:
Why is this changing?

Reply:
typed array was giving a 'invalid length' error with Uint32Array. So its needed to get it compiling. Though I should check to see if 'T' is something other than u8 if this works. Might need to be more dynamic and check the type and wrap accordingly.